core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/ResponseTransformer.java

```diff
@@ -59,12 +61,12 @@
  * <p>
  * Implementations should have proper handling of Thread interrupts. For long running, non-interruptible tasks, it is recommended
  * to check the thread interrupt status periodically and throw an {@link InterruptedException} if set. When an {@link
- * InterruptedException} is thrown from a interruptible task, you should either re-interrupt the current thread or throw that
- * {@link InterruptedException} from the {@link #apply(Object, AbortableInputStream)} method. Failure to do these things may
- * prevent the SDK from stopping the request in a timely manner in the event the thread is interrupted externally.
+ * InterruptedException} is thrown from a interruptible task, you should throw that {@link InterruptedException} from the
```

Comment:
Did we remove the ability for customers to just interrupt the thread without (re)throwing an `InterruptedException`?

Reply:
Not sure what you mean? Can we add a check for `Thread.isInterrupted()` after we call `transform`? 

To me it's perfectly valid for customers to do this in their transform methods:

```java
MyObject transform() {
    try {
        // something interruptible
    } catch (InterruptedException e) {
        log.error("Transform was interrupted", e);
        Thread.interrupt();
        return null;
    }
}
```

We have an explicit test for this behavior in the S3 module (though it's currently disabled): https://github.com/aws/aws-sdk-java-v2/blob/2.0.0-preview-12/services/s3/src/it/java/software/amazon/awssdk/services/s3/GetObjectFaultIntegrationTest.java#L108-L129

Reply:
Right, we can add the check after we call `transform`, but the time when the request actually gets timed out might be far from accurate. 

For example, say we set the request timeout to `1s` and if the `transform` method takes more than `5s`, the request only will throw `TimeoutException` after 5s when it goes to `InterruptMonitor.checkInterrupted()`, or even worse, if the request gets stuck in `transform`, TimeoutException will never get thrown.

Yeah, I can see it's valid for customers to just re-interrupt the thread, but my point is it's highly possible that they get TimeoutException long after the configured timeout.

Reply:
But we already state in the doc that customers should throw an `InterruptedException` if their transform method gets interrupted, so I guess I don't see how that's too different (as far as timing accuracy) if instead of re-throwing the exception, they set the flag and return right away and on the SDK side we have a check for the flag as soon as `transform()` returns.

Reply:
Yeah, we should be fine for most of the cases. I was thinking of one edge case where the customers just set the interrupted flag without re-throwing any exception and do some time-consuming task after it.

I will update the docs.